src/app/ReadClient.h

```diff
@@ -115,16 +141,21 @@ class ReadClient : public Messaging::ExchangeDelegate
      *
      */
     bool IsFree() const { return mState == ClientState::Uninitialized; }
+    bool IsSubscriptionTypeIdle() const { return mState == ClientState::SubscriptionActive; }
     bool IsAwaitingInitialReport() const { return mState == ClientState::AwaitingInitialReport; }
+    bool IsAwaitingSubscribeResponse() const { return mState == ClientState::AwaitingSubscribeResponse; }
+
     CHIP_ERROR GenerateEventPathList(EventPathList::Builder & aEventPathListBuilder, EventPathParams * apEventPathParamsList,
                                      size_t aEventPathParamsListSize);
     CHIP_ERROR GenerateAttributePathList(AttributePathList::Builder & aAttributeathListBuilder,
```

Comment:
SmartThings is very excited to start integrating Subscription Interactions, and this PR is an excellent path forward. 
In fact, we have started already using the interactions, by using the Read Interaction, which is based on lib/connectedhomeip/src/app/tests. However, we run into a couple of issues with the ReadClient class:

Currently ReadClient.h has GenerateAttributePathList(), ProcessReportData() and the destructor (virtual ~ReadClient()) marked as private, making them inaccessible for use outside the class. To get past this hurdle, we have declared them as public so that our application can use the ReadClient class.

I suggest setting them up as public, so that the SDK consumer can use the ReadClient class, in a similar way the TestReadInteraction (as it is set as a friend class, but this would not be a path moving forward for custom code). Or if this is not preferable, we are open to alternatives that you can suggest so Read Interaction development can be continue.